cpp/src/parquet/column_reader.h

```diff
@@ -300,6 +300,9 @@ class TypedColumnReader : public ColumnReader {
                                           int16_t* rep_levels, int32_t* indices,
                                           int64_t* indices_read, const T** dict,
                                           int32_t* dict_len) = 0;
+
+  virtual void ReadLevels(int64_t batch_size, int16_t* def_levels, int16_t* rep_levels,
```

Comment:
What do you think about making this private with a [Test Peer](https://abseil.io/tips/135) that can be used in the benchmark, so we can check this PR in?